refactor-group/refactor-platform-fe
# Coaching session fast switch select component

## Description
This PR adds a coaching session quick selector to the coaching session page.


#### GitHub Issue: None

### Changes
* Add a new coaching session quick selector component to the coaching session page
* Fix the useEffect() render loops that existed on the coaching session page
* Allow the entire button to be clicked to join a coaching session from the dashboard page

### Screenshots / Videos Showing UI Changes (if applicable)
![Jim __ Caleb _ Nov 21_ 2024](https://github.com/user-attachments/assets/0ddfb5c7-b8c8-4e55-84f3-0e2447ca223c)

### Testing Strategy
1. Navigate to a coaching session
2. Notice the new component towards the upper right hand corner
3. Select a different coaching session
4. Notice that the contents of the coaching session page change to the new session


### Concerns
None
